homeassistant/components/escea/discovery.py

```diff
@@ -17,11 +16,11 @@
 class DiscoveryService(Listener):
     """Discovery data and interfacing with pescea library."""
 
-    def __init__(self, hass):
+    def __init__(self, hass: HomeAssistant) -> None:
         """Initialise discovery service."""
         super().__init__()
-        self.hass = hass
-        self.pi_disco = None
+        self.hass: HomeAssistant = hass
+        self.pi_disco: DiscoveryService
```

Comment:
We need to initialize the instance attributes here.
```suggestion
        self.pi_disco: DiscoveryService | None = None
```